tests/unit/test_charm.py

```diff
@@ -219,3 +217,47 @@ def test_main_ingress(update, harness):
     }
 
     assert data == yaml.safe_load(relation_data["data"])
+
+
+@patch("charm.KubernetesServicePatch", lambda *_, **__: None)
+def test_dex_oidc_config_with_data(harness):
+    """Test the relation data has values by default as the charm is broadcasting them."""
+    harness.set_leader(True)
+    harness.begin()
+
+    rel_id = harness.add_relation(DEX_OIDC_CONFIG_RELATION, "app")
+    rel_data = harness.model.get_relation(DEX_OIDC_CONFIG_RELATION, rel_id).data[harness.model.app]
+
+    # Default values are expected
+    expected = {"issuer-url": f"http://{harness.model.app.name}.{harness.model.name}.svc:5556/dex"}
+    assert rel_data["issuer-url"] == expected["issuer-url"]
+
+
+@patch("charm.KubernetesServicePatch", lambda *_, **__: None)
+def test_grpc_relation_with_data_when_data_changes(
+    harness,
+):
+    """Test the relation data on config changed events."""
+    harness.set_leader(True)
+    # Change the configuration option before starting harness so
+    # the correct values are passed to the K8sServiceInfo lib
+    # FIXME: the correct behaviour should be to change the config
+    # at any point in time to trigger config events and check the
+    # value gets passed correctly when it changes.
```

Comment:
I remember this discussion but do we have an issue for this? I mean if at some point there is a fix, we could link this.

Reply:
We do not have an issue, but also I'm not sure if there is a fix as this would be harness' behaviour. I can create one in our tracker just so we remember what it is and what are the things we have to do.

Reply:
Let's if there is an issue in Harness. Otherwise, it's fine as is.